src/exec/noexec.rs

```diff
@@ -281,11 +361,41 @@ pub(crate) fn add_noexec_filter(command: &mut Command) -> SpawnNoexecHandler {
             // SAFETY: seccomp_data can be safely zero-initialized.
             let dummy: seccomp_data = zeroed();
             let nr_offset = (&dummy.nr) as *const _ as usize - &dummy as *const _ as usize;
+            let arch_offset = (&dummy.arch) as *const _ as usize - &dummy as *const _ as usize;
 
             // SAFETY: libc unnecessarily marks these functions as unsafe
-            let exec_filter: [sock_filter; 5] = [
+            #[rustfmt::skip]
+            let exec_filter = [
```

Comment:
I remember why I had the type explicitly specified: I wanted to ensure that the element type matches what we cast the pointer to when creating the `sock_fprog`, but using `_` as array length is unstable.

Reply:
I had a suspicion that was the case, although I feel comfortable enough with Rust to 'see' that this type is an array, not a slice. But let me create a work-around for you and see if you like that.

Reply:
See 04dcdca